go-controller/pkg/node/gateway_init_linux_test.go

```diff
@@ -288,6 +295,20 @@ func shareGatewayInterfaceTest(app *cli.App, testNS ns.NetNS,
 				LinkIndex: l.Attrs().Index,
 				Src:       ifAddrs[0].IP,
 			}
+			Eventually(func() error {
+				r, err := util.LinkRouteGetFilteredRoute(
+					expRoute,
+					netlink.RT_FILTER_DST|netlink.RT_FILTER_OIF|netlink.RT_FILTER_SRC,
+				)
+				if err != nil {
+					return err
+				}
+				if r == nil {
+					return fmt.Errorf("failed to find route")
+				}
+				return nil
+			}, 1*time.Second).ShouldNot(HaveOccurred())
+
 			route, err := util.LinkRouteGetFilteredRoute(
```

Comment:
do we need to keep this part if you added the same check in eventually?

Reply:
removed